### PR TITLE
refactor: Prepare for including a NS with Ident

### PR DESCRIPTION
**Pull request by max-sixty**
This adds an additional Struct while we move over; so we don't need to do one huge PR with the merge conflicts that entails

ref #1031

I'm adding this to the semantic branch to avoid future merge conflicts, even though I'm not a great fan of abandoning main (I also wanted to do _something_ this weekend and wasn't making enough progress with the "onboarding" issues, so chose this)

While they both exist, the only tax is some `.into()` calls

### CODE REVIEW

File: prql-compiler/src/ast/expr.rs [hunk 1]
```diff
@@ -60,6 +60,46 @@ pub enum ExprKind {
 /// A name. Generally columns, tables, functions, variables.
 pub type Ident = String;
 
+// We are moving `Ident` from being a string to containing the structure of a
+// namespace. Eventually we can remove this and just use `Ident` everywhere.
+// https://github.com/prql/prql/issues/1031
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
+pub struct IdentWithNamespace {
+    pub namespace: Option<String>,
+    pub ident: Ident,
```
Comment (aljazerzen) [inline prql-compiler/src/ast/expr.rs:69]: I think we need a better name for this. Ident is the whole node, an identifier, while this field is the specific name in a namespace. I used "variable name" or "var_name", but I'm open to suggestions.

File: prql-compiler/src/ast/expr.rs [hunk 2]
```diff
@@ -60,6 +60,46 @@ pub enum ExprKind {
 /// A name. Generally columns, tables, functions, variables.
 pub type Ident = String;
```
Comment (aljazerzen) [inline prql-compiler/src/ast/expr.rs:61]: I believe that this can be fully replaced by IdentWithNamespace. Because all idents can have namespace, right?

**Review by aljazerzen — APPROVED** [review]
Nice integration with current ident handling and well prepared for improved ident parsing.

File: prql-compiler/src/ast/expr.rs (hunk 2 above)
Comment (max-sixty) [reply prql-compiler/src/ast/expr.rs:61]: Yes, my plan was to move it incrementally; then probably rename `IdentWithNamespace` back to `Ident`. I wanted to do the closest thing first, so we didn't get conflicts etc. (But maybe it's not that hard and we could do it in one fell swoop...)

File: prql-compiler/src/ast/expr.rs (hunk 1 above)
Comment (max-sixty) [reply prql-compiler/src/ast/expr.rs:69]: Yes agree. `var_name` is fine; how about just `var`?

**Comment by max-sixty** [comment]
Merging to ensure we don't get more conflicts — will continue iterating on it!

File: prql-compiler/src/ast/expr.rs (hunk 1 above)
Comment (aljazerzen) [reply prql-compiler/src/ast/expr.rs:69]: `var` it is

File: prql-compiler/src/ast/expr.rs (hunk 2 above)
Comment (aljazerzen) [reply prql-compiler/src/ast/expr.rs:61]: That's my philosophy on semantic branch - just do it in one fell swoop :D

Comment (max-sixty) [reply prql-compiler/src/ast/expr.rs:61]: Ha! We are so aligned on so many dimensions; "average PR size" is the outlier!

**Pull request merged**